packages/overlay/src/Overlay.ts

```diff
@@ -95,13 +101,23 @@ export class Overlay extends OverlayFeatures {
      * provided that option.
      */
     @property({ type: Boolean })
-    override delayed = false;
+    override get delayed(): boolean {
+        if (this.elements.length === 0) {
+            return false;
+        }
 
-    @query('.dialog')
-    override dialogEl!: HTMLDialogElement & {
-        showPopover(): void;
-        hidePopover(): void;
-    };
+        if (this.elements[0].getAttribute('delayed') !== null) {
+            return true;
+        }
+
+        return this._delayed;
```

Comment:
Is this all enough the same as `return this.elements.at(-1)?.hasAttribute('delayed') ?? this._delayed;`?

Reply:
Almost, changed to `this.elements.at(-1)?.hasAttribute('delayed') || this._delayed;`